### ISSUE TITLE
Glob Exclude entries in .rubocop_todo.yml are silently skipped by --report-unused-todo-entries

### ISSUE THREAD

**Issue by yunusdim** [rubocop/rubocop#15570]
Hi, an observation about `--report-unused-todo-entries` (`TodoAudit`), not a bug report. It comes from a formal result on separation criteria I deposited recently.

**The result, briefly.**

A contract of the form "these cops must not fire here, except what's declared excluded" is a separation criterion evaluated against a declaration of what's permitted. Such criteria have a general degeneration: if the permitted set can be enlarged, every violation can be removed by enlarging it, without touching the code, and the required enlargement is computable from the violations themselves. What blocks the degeneration is granularity, and what makes granularity enforceable is actually auditing every entry, not just the ones that happen to resolve to a literal path.

**Where the gap is.**

`TodoAudit#offending_cops_without_todo` builds its audit set like this:

```ruby
files = entries.map { |entry| absolute(entry.path) }.uniq.select { |file| File.file?(file) }
```

`Exclude` entries are strings from the YAML, RuboCop's own `Exclude` key accepts globs, and several cops (mostly the RSpec family) ship a glob in their default `Exclude`, which `--auto-gen-config` then folds into `.rubocop_todo.yml` as-is. `File.file?` is false for a glob string regardless of whether files matching it exist, so any glob entry never enters `files`, and the audit never inspects it, never flags it as stale, never checks whether the set of files it silently permits has grown.

I reproduced the filtering directly (it's plain Ruby, no lint engine involved): a literal entry for an existing file passes `File.file?` and gets audited; `**/*_spec.rb`, with a real matching file present on disk, doesn't, and is dropped before the audit sees it.

This isn't a rare shape. A code search across `.rubocop_todo.yml` files turns up roughly 6,300 of them with a glob in some `Exclude`, `spec/**/*`, `**/*_spec.rb`, and similar, in repos as large as `mongodb/mongoid` and `gocardless/statesman`. Mostly because RSpec cops default to excluding specs by glob, so the pattern arrives in generated todo files without anyone hand-writing it.

**Concrete question.**

Would you consider expanding glob `Exclude` entries against the project's file list (`Dir.glob` relative to the project root, the same expansion RuboCop already does when applying `Exclude` during a real run) before the `File.file?` filter, so `TodoAudit` can audit what those entries actually cover instead of skipping them outright?

Reference, if the argument is of interest: *Closure Trivialization in Typed Separation Systems*, https://doi.org/10.5281/zenodo.21908527, §10 makes the granularity distinction precise.
